### PR TITLE
angular 8 upgrade

### LINKED ISSUE #708: ng8 upgrade

### PR DESCRIPTION
**Pull request by sandikbarr**
closes #708

### CODE REVIEW

File: src/app/shared/wysiwyg/prosemirror.d.ts [hunk 1]
```diff
@@ -1,149 +1,3 @@
...
-  export { orderedListRule };
-  function bulletListRule(nodeType: any): InputRule;
-  export { bulletListRule };
-  function codeBlockRule(nodeType: any): InputRule;
-  export { codeBlockRule };
-  function headingRule(nodeType: any, maxLevel: any): InputRule;
-  export { headingRule };
-}
-
 declare module 'prosemirror-markdown' {
```
Comment (sandikbarr) [inline src/app/shared/wysiwyg/prosemirror.d.ts:1]: for some reason I don't understand, it doesn't see the exported schema from @types/prosemirror-markdown, so I had to keep just our type defs for prosemirror-markdown

File: src/app/shared/wysiwyg/prosemirror.markdown.editor.ts [hunk 2]
```diff
@@ -55,63 +60,67 @@ export class ProseMirrorMarkdownEditor {
...
+      doc = DOMParser.fromSchema(basicSchema).parse(domNode);
     } else {
-      config['doc'] = defaultMarkdownParser.parse(this.content ? this.content : '');
-    }
-    return config;
+      doc = defaultMarkdownParser.parse(this.content ? this.content : '');
+    }
+    return {
+      doc,
+      plugins: [
```
Comment (sandikbarr) [inline src/app/shared/wysiwyg/prosemirror.markdown.editor.ts:115]: * inputRules: the types of marks accepted by the schema. [This editor supports both markdown (description) and html (itunes summary) output.]
* keymap: shortcut key mappings
* menuBar: plugin containing menu items and dropdown menus
* history: \*new\* adds cmd+z undo/redo support
* class attribute for prosemirror CSS

File: src/app/story/directives/status-control.component.ts [hunk 3]
```diff
@@ -1,28 +1,37 @@
 import { Component, DoCheck, Input, Output, EventEmitter, ViewChild, ElementRef } from '@angular/core';
```
Comment (sandikbarr) [inline src/app/story/directives/status-control.component.ts:1]: this is just a lot of auto formatting with the ViewChild update, but it did actually catch some `,`/`;` syntax issues 😬

File: src/testing/index.ts [hunk 4]
```diff
@@ -16,7 +16,9 @@ export { findComponent, niceEl } from './helpers';
...
 // HACKY: export a getter to the current cms-mocks instance
-export const cms: MockHalService = null;
-Object.defineProperty(module.exports, 'cms', {
-  get: () => currentHal() || new MockHalService()
-});
+// export const cms: MockHalService = null;
+// Object.defineProperty(module.exports, 'cms', {
+//   get: () => currentHal() || new MockHalService()
+// });
+// ^ above hack stopped working, so...?
```
Comment (sandikbarr) [inline src/testing/index.ts:23]: hack stopped working =/

What this does is get the `cms` service that was provided in the TestBed so we can mock hal api calls off it. I changed it to a function call that usually happens in each test's beforeEach, did the trick.

File: src/tsconfig.spec.json [hunk 5]
```diff
@@ -2,20 +2,10 @@
...
   "compilerOptions": {
     "outDir": "../out-tsc/spec",
-    "module": "commonjs",
-    "target": "es5",
     "baseUrl": "",
-    "types": [
-      "jasmine",
-      "node"
-    ]
+    "target": "es5",
```
Comment (sandikbarr) [inline src/tsconfig.spec.json:6]: `target: es5` is a fix for karma/phantom test runner failing with `Use of reserved word 'class'`

File: src/app/story/directives/status-control.component.ts (hunk 3 above)
Comment (cavis) [reply src/app/story/directives/status-control.component.ts:1]: Nice to have this stuff linting, though.  But you are reminding me of all my poor syntax decisions in this repo ... i just love trying to fit everything on 1 line in the `template`!

File: src/app/shared/wysiwyg/prosemirror.markdown.editor.ts [hunk 6]
```diff
@@ -55,63 +60,67 @@ export class ProseMirrorMarkdownEditor {
...
+        inputRules({ rules: this.buildInputRules(this.outputSchema) }),
+        keymap(this.buildKeymap(this.outputSchema)),
+        keymap(baseKeymap),
+        menuBar({ floating: true, content: this.buildMenuItems(this.outputFormat, this.outputSchema, this.images) }),
+        history(),
+        new Plugin({ props: { attributes: { class: 'ProseMirror-example-setup-style' } } })
```
Comment (cavis) [inline src/app/shared/wysiwyg/prosemirror.markdown.editor.ts:121]: Was going to ask if you meant to leave this in ... but i see it was already in here!

File: src/app/shared/wysiwyg/prosemirror.markdown.editor.ts (hunk 2 above)
Comment (cavis) [reply src/app/shared/wysiwyg/prosemirror.markdown.editor.ts:115]: Nice!  Seems like some decent improvements.

File: src/app/search/search.component.spec.ts [hunk 7]
```diff
@@ -28,17 +27,16 @@ describe('SearchComponent', () => {
...
+    const params = { tab: 'stories', perPage: 12, orderBy: 'published_at', orderDesc: false, page: 1 };
     spyOn(comp.router, 'navigate');
     comp.searchWithParams(params);
     expect(comp.router.navigate).toHaveBeenCalledWith(['search', params]);
   });
 
   describe('on the stories tab', () => {
-
-    let storiesResults = [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}];
+    const storiesResults = new Array(15).fill({});
```
Comment (cavis) [inline src/app/search/search.component.spec.ts:37]: Nice.  As much as i like the manually-defining-15-things.

**Pull request merged**